### PR TITLE
Generate memref.assert_alignment ops

### LINKED ISSUE #8293: IREEComprehensiveBufferize does not generate memref.assume_alignment
We do want generate the alignment op in IREE. IREE does not generate the op after switching to IREEComprehensiveBufferize. File an issue for tracking it.

### PR DESCRIPTION
**Pull request by matthias-springer**
This fixes #8293.

### CODE REVIEW

File: iree/compiler/Codegen/Common/test/iree_comprehensive_bufferize.mlir [hunk 1]
```diff
@@ -40,7 +40,8 @@ func @matmul() {
 //  CHECK-DAG:   %[[M:.+]] = hal.interface.constant.load[0]
 //  CHECK-DAG:   %[[N:.+]] = hal.interface.constant.load[1]
 //  CHECK-DAG:   %[[K:.+]] = hal.interface.constant.load[2]
-//  CHECK-DAG:   %[[LHS:.+]] = hal.interface.binding.subspan set(0) binding(0) type(storage_buffer)
+//  CHECK-DAG:   %[[LHS:.+]] = hal.interface.binding.subspan set(0) binding(0) type(storage_buffer) alignment(32)
+//      CHECK:   memref.assume_alignment %[[LHS]], 32
```
Comment (hanhanW) [inline iree/compiler/Codegen/Common/test/iree_comprehensive_bufferize.mlir:44]: Maybe CHECK-DAG could fix the issue

Comment (matthias-springer) [reply iree/compiler/Codegen/Common/test/iree_comprehensive_bufferize.mlir]: Turns out I put changed two different test case ;)

Comment (matthias-springer) [reply iree/compiler/Codegen/Common/test/iree_comprehensive_bufferize.mlir]: Turns out I put changed two different test case ;)

File: iree/compiler/Codegen/Common/test/iree_comprehensive_bufferize.mlir [hunk 2]
```diff
@@ -114,7 +114,8 @@ func @matmul_fill() {
 //  CHECK-DAG:   %[[M:.+]] = hal.interface.constant.load[0]
 //  CHECK-DAG:   %[[N:.+]] = hal.interface.constant.load[1]
 //  CHECK-DAG:   %[[K:.+]] = hal.interface.constant.load[2]
-//  CHECK-DAG:   %[[LHS:.+]] = hal.interface.binding.subspan set(0) binding(0) type(storage_buffer)
+//  CHECK-DAG:   %[[LHS:.+]] = hal.interface.binding.subspan set(0) binding(0) type(storage_buffer) alignment(32)
```
Comment (MaheshRavishankar) [inline iree/compiler/Codegen/Common/test/iree_comprehensive_bufferize.mlir:117]: Is there a reason the other interface bindings dont get the alignment?

**Review by MaheshRavishankar — CHANGES_REQUESTED** [review]

File: iree/compiler/Codegen/Common/test/iree_comprehensive_bufferize.mlir (hunk 2 above)
Comment (matthias-springer) [reply iree/compiler/Codegen/Common/test/iree_comprehensive_bufferize.mlir:117]: The alignment was not specified in the in the test case. It is only specified for this one.

**Review by MaheshRavishankar — APPROVED** [review]

**Pull request merged**